src/main/java/io/vlingo/symbio/store/object/jdbc/JDBCObjectStoreEntryReaderActor.java

```diff
@@ -150,9 +150,20 @@ public void close() {
       entriesQuery.setLong(2, offset + maximumEntries - 1L);
       try (final ResultSet result = entriesQuery.executeQuery()) {
         final List<Entry<String>> entries = mapQueriedEntriesFrom(result);
-        offset += entries.size();
-        updateCurrentOffset();
-        return completes().with(entries);
+        List<Long> gapIds = reader().detectGaps(entries, offset, maximumEntries);
```

Comment:
@pflueras This is all good! I was thinking that this common code from both primary read methods could be put in one place. Probably all of the JDBC `EntryReader` will use this. WDYT? It seems it would save you a lot of time on the other implementations to reuse. I assume you have tests already to ensure that the refactoring is correct.

Reply:
Thank you! I tried to extract as much as possible common/generic code in `GapRetryReader` class (vlingo-symbio). Remaining code looks like similar across `EntryReader`s but it is quite specific for each individual reader.